### PR TITLE
SRE2-1549: Adds multi-region kms key module

### PR DESCRIPTION
**Pull request by rafops**
- Added `kms_multi_region_key` to be used initially by `helm_secrets` module to encrypt/decrypt secrets using a KMS key that is highly available across primary and replica regions

### CODE REVIEW

File: modules/kms_multi_region_key/outputs.tf [hunk 1]
```diff
@@ -0,0 +1,7 @@
+output "kms_key_arn" {
+  value = aws_kms_key.primary.arn
+}
+
+output "kms_key_replica_arn" {
```
Comment (joeauty) [inline modules/kms_multi_region_key/outputs.tf:5]: why is this needed?

**Review by joeauty — CHANGES_REQUESTED** [review]

File: modules/kms_multi_region_key/main.tf [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+  policy = data.aws_iam_policy_document.policy.json
+}
+
+resource "aws_kms_alias" "kms_alias" {
+  name          = "alias/${var.environment_name}"
+  target_key_id = aws_kms_key.primary.key_id
+}
+
+resource "aws_kms_replica_key" "replica" {
```
[the remark this answers is not in the document]
Comment (rafops) [reply modules/kms_multi_region_key/main.tf:96]: This is to ensure there's a replica of the primary key available in the replica region. `multi_region = true` is not enough.

File: modules/kms_multi_region_key/outputs.tf (hunk 1 above)
Comment (rafops) [reply modules/kms_multi_region_key/outputs.tf:5]: You can have a replica key as a backup when configuring sops:
```
---
creation_rules:
- path_regex: \.test\.yaml$
  kms: arn:aws:kms:us-east-1:021589213323:key/mrk-fa874364e1964213a6827776fcdb472f
  # kms: arn:aws:kms:us-west-2:021589213323:key/mrk-fa874364e1964213a6827776fcdb472f
  aws_profile: nulogy-go-test
```

**Review by joeauty — APPROVED** [review]
